imap_processing/glows/l1b/glows_l1b.py

```diff
@@ -114,28 +221,31 @@ def process_histogram(l1a: xr.Dataset) -> xr.Dataset:
 
     # This preserves the dimensions for the histogram output
     new_dims[0] = ["bins"]
-    new_dims[22] = ["bins"]  # For imap_spin_angle_center - a per-bin value
+    new_dims[21] = ["bins"]  # For imap_spin_angle_center - a per-bin value
```

Comment:
This is pretty cryptic and seems easy to get off-by-one errors (for example, I am unclear why this has changed all of a sudden now). Is there a way we can have `new_dims` be string lookup instead of index-based?

Reply:
`new_dims` must be index based, but I can write a lookup method using the dataclass to generate it.

Reply:
Rewrote it and I think it's way better, actually!